tty/usbacm/usbacm.c

```diff
@@ -675,12 +631,23 @@ static int usbacm_handleInsertion(usb_driver_t *drv, usb_devinfo_t *insertion, u
 		return usb_modeswitchHandle(drv, insertion, mode);
 	}
 
-	if ((dev = usbacm_devAlloc()) == NULL)
+	dev = usbacm_devAlloc();
+	if (dev == NULL) {
 		return -ENOMEM;
+	}
 
 	dev->instance = *insertion;
 	dev->drv = drv;
 
+	mutexLock(usbacm_common.lock);
+
+	err = idtree_alloc(&usbacm_common.devices, &dev->node);
```

Comment:
Consider moving this section to `usbacm_devAlloc`, similar to how it's handled in umass driver. You could invoke `_usbacm_devAlloc` while holding the mutex.